File: front/apps/web-ele/src/api/core/purchase_order.ts
import { requestClient } from '../request';

export interface PurchaseOrder {
  id: number;
  purchaseCode: string;
  orderCode: string;
  supplierCode: string;
  purchaseDate: string;
  purchaseStatus: string;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
}

export interface PurchaseOrderListResponse {
  total: number;
  list: PurchaseOrder[];
}

export async function createPurchaseOrder(data: {
  purchaseCode: string;
  orderCode: string;
  supplierCode: string;
  purchaseDate: string;
  purchaseStatus: string;
  createdBy: number;
}) {
  const response = await requestClient.post<PurchaseOrder>('/purchase_orders', data);
  return response;
}

export async function getPurchaseOrders(params?: { limit?: number; offset?: number }) {
  const response = await requestClient.get<PurchaseOrderListResponse>('/purchase_orders', {
    params: {
      limit: params?.limit || 100,
      offset: params?.offset || 0,
    },
  });
  return response;
}

export async function getPurchaseOrder(id: number) {
  const response = await requestClient.get<PurchaseOrder>(`/purchase_orders/${id}`);
  return response;
}

export async function updatePurchaseOrder(
  id: number,
  data: {
    purchaseCode?: string;
    orderCode?: string;
    supplierCode?: string;
    purchaseDate?: string;
    purchaseStatus?: string;
  }
) {
  const response = await requestClient.post<PurchaseOrder>(`/purchase_orders/${id}`, data);
  return response;
}

export async function deletePurchaseOrder(id: number) {
  const response = await requestClient.delete(`/purchase_orders/${id}`);
  return response;
}

File: back/internal/purchase_order/domain/purchase_order.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// PurchaseOrder 采购单
type PurchaseOrder struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	PurchaseCode string `gorm:"size:50;uniqueIndex;not null" json:"purchaseCode"`
	OrderCode string `gorm:"size:50;not null;index" json:"orderCode"`
	SupplierCode string `gorm:"size:50;not null;index" json:"supplierCode"`
	PurchaseDate time.Time `gorm:"type:date;not null" json:"purchaseDate"`
	PurchaseStatus string `gorm:"size:50;not null;default:'pending'" json:"purchaseStatus"`
	CreatedBy uint           `gorm:"not null" json:"createdBy"`
	CreatedAt time.Time      `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt time.Time      `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

// TableName 表名
func (PurchaseOrder) TableName() string {
	return "purchase_orders"
}
